src/Integration.Vsix/Analysis/DisableRuleCommand.cs

```diff
@@ -156,15 +155,15 @@ private void Execute(object sender, EventArgs e)
             SonarRuleRepoKeys.TypeScript
         };
 
-        private static void CalculateStatuses(SonarCompositeRuleId rule, SonarLintMode mode, out bool isVisible, out bool isEnabled)
+        private static void CalculateStatuses(SonarCompositeRuleId rule, SonarLintMode mode, bool isConnectedModeSupportedForSecrets, out bool isVisible, out bool isEnabled)
         {
             // Special case: Secrets
-            // We don't support connected mode for Secrets at the moment so we allow disabling
-            // secrets rules in connected mode.
+            // We don't support connected mode for Secrets in SonarQubeVersions < 9.9 so we allow disabling
+            // secrets rules in those cases.
             if (SonarRuleRepoKeys.AreEqual(SonarRuleRepoKeys.Secrets, rule.RepoKey))
             {
                 isVisible = true;
-                isEnabled = true;
+                isEnabled = !isConnectedModeSupportedForSecrets;
```

Comment:
I realize it is a bit confusing to use the negative here, hence why i made the boolean name so long . However I dont think it makes it any less confusing

Reply:
Sometimes it's difficult to find a good name or avoid a negation. You've added a comment and the name is descriptive, so that's probably as much as we can do.

An alternative approach would be to make the method non-`static`, and perhaps create a separate function for the secrets check and explanatory comment. That avoids having to think of a variable name for the `isConnectedModeSupportedForSecrets` boolean e.g.

```
        private void CalculateStatuses(SonarCompositeRuleId rule, out bool isVisible, out bool isEnabled)
        {
            // Special case: Secrets
            if (SonarRuleRepoKeys.AreEqual(SonarRuleRepoKeys.Secrets, rule.RepoKey))
            {
                CalculateStatusesForSecrets(out isVisible, out isEnabled);
                return;
            }

            // We don't currently support disabling rules for all repos
            if (supportedRepos.Contains(rule.RepoKey, SonarRuleRepoKeys.RepoKeyComparer))
            {
                isVisible = true;
                isEnabled = activeSolutionBoundTracker.CurrentConfiguration.Mode == SonarLintMode.Standalone;
                return;
            }

            isVisible = false;
            isEnabled = false;
        }

        private void CalculateStatusesForSecrets(out bool isVisible, out bool isEnabled)
        {
            isVisible = true;

            // We don't support connected mode for Secrets in SonarQubeVersions < 9.9 so we
            // still allow users to disable secrets rules in those cases.
            isEnabled = !connectedModeSecrets.AreSecretsAvailable();
        }
```

Reply:
Another approach is to refactor the code to avoid out parameters:

```
        private void QueryStatus(object sender, EventArgs args)
        {
            try
            {
                var isVisible = false;
                var isEnabled = false;
                if (errorListHelper.TryGetRuleIdFromSelectedRow(out var ruleId))
                {
                    isVisible = IsSonarRule(ruleId);
                    isEnabled = isVisible && IsDisablingRuleAllowed(ruleId);
                }

                menuItem.Visible = isVisible;
                menuItem.Enabled = isEnabled;
            }
            catch (Exception ex) when (!ErrorHandler.IsCriticalException(ex))
            {
                logger.WriteLine(AnalysisStrings.DisableRule_ErrorCheckingCommandStatus, ex.Message);
            }
        }

        private bool IsSonarRule(SonarCompositeRuleId rule)
            => supportedRepos.Contains(rule.RepoKey, SonarRuleRepoKeys.RepoKeyComparer);

        private bool IsDisablingRuleAllowed(SonarCompositeRuleId rule)
        {
            // Special case: secrets
            // We don't support connected mode for Secrets in SonarQubeVersions < 9.9 so we
            // still allow users to disable secrets rules in those cases.
            if (SonarRuleRepoKeys.AreEqual(SonarRuleRepoKeys.Secrets, rule.RepoKey))
            {
                return !connectedModeSecrets.AreSecretsAvailable();
            }

            // Otherwise, can only disables rules in standalone mode
            return activeSolutionBoundTracker.CurrentConfiguration.Mode == SonarLintMode.Standalone;
        }
```

That's probably simplest. We have enough tests coverage, so changing the methods completely like this is fine if we want to.

Reply:
Really like the second option, much more elegant!